http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/BootstrapProvider.java

```diff
@@ -56,8 +61,21 @@ public Bootstrap createBootstrap(String host, int port) {
                 .option(ChannelOption.CONNECT_TIMEOUT_MILLIS, nettyConfiguration.connectTimeoutMillis())
                 .option(ChannelOption.SO_KEEPALIVE, nettyConfiguration.tcpKeepAlive())
                 .remoteAddress(InetSocketAddress.createUnresolved(host, port));
+
+        if (nettyConfiguration.isNonBlockingResolver()) {
+            bootstrap.resolver(nonBlockingResolverGroup());
+        }
+
         sdkChannelOptions.channelOptions().forEach(bootstrap::option);
 
         return bootstrap;
     }
+
+    private AddressResolverGroup<InetSocketAddress> nonBlockingResolverGroup() {
+        DnsNameResolverBuilder builder = new DnsNameResolverBuilder()
+            .channelType(NioDatagramChannel.class)
+            .resolveCache(NoopDnsCache.INSTANCE);
```

Comment:
What's the reasoning for a no-op cache, instead of the default behavior?

Reply:
Ah, interesting. I think the existing DNS resolver uses the JVM's DNS cache. I don't know the context for that requirement, let me investigate further...

Reply:
It looks like that comment is referring to a bug we had in the past that seems unrelated to this.

The bug was that we'd pass a _resolved_ RemoteAddress to netty here: https://github.com/aws/aws-sdk-java-v2/blob/master/http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/BootstrapProvider.java#L58

This meant that any requests to the same hostname wouldn't have its DNS resolved by Netty at all. The comment was trying to prevent us from resolving the address ourselves before passing it to Netty, so that Netty would do the resolution.

I think we can still enable the DNS cache in this case, since it's essentially replacing the JVM DNS cache in the old code path, not the "cached" DNS lookup of the original bug.

Reply:
Done.